abcclassroom/dirsetup.py

```diff
@@ -0,0 +1,94 @@
+"""
+abc-classroom.dir-setup
+========================
+
+"""
+
+import os
+import shutil
+import argparse
+import abcclassroom
+
+
+def path_to_example(dataset):
+    """
+    Gets the path to an example dataset.
+
+    Parameters
+    ----------
+    dataset : string
+        stringy boi
+    """
+    abcclassroom_path = os.path.split(abcclassroom.__file__)[0]
+    data_dir = os.path.join(abcclassroom_path, "example-data")
+    data_files = os.listdir(data_dir)
+    if dataset not in data_files:
+        raise KeyError(dataset + " not found in abc-classroom example data.")
+    return os.path.join(data_dir, dataset)
+
+
+def directory_setup():
+    """
+    Construct a file path to an example dataset.
+    This file defines helper functions to access data files in this directory,
+    to support examples. Adapted from the PySAL package.
+
+    Parameters
+    ----------
+    dataset: string
+        Name of a dataset to access (e.g., "epsg.json", or "RGB.byte.tif")
+
+    Returns
+    -------
+        A file path (string) to the dataset
+    """
+    # Creating the default names for the cloned and template repos.
+    cloned = "cloned_repos"
+    template = "template_repos"
+    course = "course_dir"
+    # Making sure the configuration file is where it's supposed to be.
+    config = path_to_example("sample_config.yml")
+    if not os.path.exists(config):
+        raise ValueError(
+            "Configuration file can't be located, please ensure abc-classroom has been installed correctly"
+        )
+    # Allows users to rename the cloned and template repos.
+    parser = argparse.ArgumentParser(description=directory_setup.__doc__)
+    parser.add_argument(
```

Comment:
There are 3 required command line arguments (`course_repo`, `template_dir`, `cloned_repo`).  The current version of the sample_config has 9 top-level options (`organization`,  `course_directory`, `course_name`, `short_coursename`, `roster`, `nbgrader_dir`, `clone_dir`, `template_dir`, and `extra_files`). The choice of which ones are available here vs only by editing the config seems arbitrary. 

I suggest keeping the command line args to a minimum and assuming default values where possible (minimum is information about the GitHub organization and the course name).

Reply:
A few notes.
I don't think the arguments are arbitrary. This particular function creates 3 folders. These three arguments define the name of the three directories that are created so they are very specific to the precise implementation of the command. This seems to make sense because a user might want to have specific directory names. The other items `roster`, `nbgrader_dir` etc are actually not related to this particular call in any way. For example nbgrader_dir is option and created via nbgrader commands. 

  * I don't think the arguments should be required however. They should be optional!

I think the bigger question is: DO we want to allow users to customize the name of the `template_repo/` and the `cloned_repo/` directory. I could be convinced that those two names if we chose them well, could be consistent and thus there are no arguments for those two directory names.

Now, if the intent is that this command also sets up the config then i could be convinced that we need more agruments but i think that is actually too complicated. i'd like to focus this command on just setting up a basic structure for the users.

Reply:
i've seen this done (github pages templates) where the config "ships" with a bunch of things commented out. i think that could work really well here so we don't think about this script as modifying much in the config file if anything. maybe just the course name.

Reply:
Good question - maybe it is not necessary to allow the user to change the names of those directories (which would be similar to how nbgrader works). That would simplify things quite a bit. 

If we change this to a `quickstart` rather than `dir-setup` command, then it would be good to capture the GitHub organization name as a command line argument, as we need that from the user (cannot use a default value). Otherwise, they _have_ to edit the config after running `quickstart`.

Reply:
ok! i love the idea of changing it to quickstart!! so let's do that.
2. it sounds like simpler is better now so @nkorinek let's NOT allow them to customize the two subdirectories for the time being (but maybe it can be something we add as an enhancement in the future. 

@kcranston what would you like this function to take as arguments then? course name and org name? it seems like the course name should define the name of the parent directory. 

the only other option would be
1. they make the directory for the course, and add a config file with variables and we then use quickstart to build everything out around that. but that seems like a bit more work for them. and we have a model with nbgrader. thoughts?

Reply:
if we have course_name and organization as args, then the user can run quickstart and not have to do anything else before using abc-classroom, so I like that.